transport/ws/src/main/java/org/kaazing/gateway/transport/ws/extension/WebSocketExtensionFactorySpi.java

```diff
@@ -53,7 +54,48 @@
      *                   (protocol violation). Throwing this exception will result in failing the WebSocket 
      *                   connection.
      */
-    public abstract WebSocketExtensionSpi negotiate(ExtensionHeader requestedExtension, WsResourceAddress address)
+    public abstract WebSocketExtension negotiate(ExtensionHeader requestedExtension, WsResourceAddress address)
             throws ProtocolException;
+    
+
+    /**
+     * This method allows extensions to specify the order that they would like to be placed on the Filter Chain.
+     *
+     * @return A {@link WebSocketExtensionFactorySpi.ExtensionOrderCategory} that specifies the extension type and where it
+     * should be ordered
+     */
+    public ExtensionOrderCategory getOrderCategory() {
+        return ExtensionOrderCategory.OTHER;
+    }
+
+    /**
+     * An ordered enum that describes the category that an extension requests to be ordered in.  The enum is ordered by the order
+     * they appear in the WebSocket handshake response extensions header.  That is the nearest to the network being last, see
+     * https://tools.ietf.org/html/draft-ietf-hybi-permessage-compression-21#section-3
+     */
+    enum ExtensionOrderCategory{
+        /**
+         * Other, meaning not any of the other categories
+         */
+        OTHER,
+        /**
+         * Extensions which transmit websocket binary or text frames which are not application data (typically these
+         * are extension-specific control frames starting with four extension-specific control bytes). The extension must consume
+         * these frames when they are being received, or transform them into standard websocket control frames, so that prior
+         * extensions (in category OTHER) and the application do not see them.
+         */
+        EMULATION,
+        /**
+         * Network, meaning something that transforms the network bytes
+         */
+        NETWORK,
+    }
+
+    @Override
+    public int compareTo(WebSocketExtensionFactorySpi o) {
+        List<ExtensionOrderCategory> extensionOrderValues =
+                Arrays.asList(WebSocketExtensionFactorySpi.ExtensionOrderCategory.values());
+        return extensionOrderValues.indexOf(getOrderCategory()) - extensionOrderValues.indexOf(o.getOrderCategory());
```

Comment:
I think you can just use the enum.ordinal method, so replace the entire method body with
`````
return getOrderCategory().ordinal() - o.getOrderCategory().ordinal();
`````